xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -107,111 +125,39 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-         * matter.
-         */
-        @Override
-        public String getPolicyName() {
-          return "xds_child_policy_balancer_name_" + newBalancerName;
-        }
+    resolvedAddresses = resolvedAddresses.toBuilder()
+        .setAttributes(attributes.toBuilder().discard(ATTR_LOAD_BALANCING_CONFIG).build())
+        .build();
+    switchingLoadBalancer.handleResolvedAddresses(resolvedAddresses);
 
-        @Override
-        public LoadBalancer newLoadBalancer(final Helper helper) {
-          return new LoadBalancer() {
-            @Nullable
-            XdsClient xdsClient;
-            @Nullable
-            LocalityStore localityStore;
-            @Nullable
-            LoadReportClient lrsClient;
-
-            @Override
-            public void handleNameResolutionError(Status error) {}
-
-            @Override
-            public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-              if (xdsClient == null) {
-                ManagedChannel channel = initLbChannel(helper, newBalancerName, channelCredsList);
-                xdsClient  = new XdsComms2(
-                    channel, helper, new ExponentialBackoffPolicy.Provider(),
-                    GrpcUtil.STOPWATCH_SUPPLIER, node);
-                LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
-                localityStore = localityStoreFactory.newLocalityStore(
-                    helper, lbRegistry, loadStatsStore);
-                // TODO(zdapeng): Use XdsClient to do Lrs directly.
-                lrsClient = loadReportClientFactory.createLoadReportClient(
-                    channel, helper, new ExponentialBackoffPolicy.Provider(),
-                    loadStatsStore);
-                final LoadReportCallback lrsCallback =
-                    new LoadReportCallback() {
-                      @Override
-                      public void onReportResponse(long reportIntervalNano) {
-                        localityStore.updateOobMetricsReportInterval(reportIntervalNano);
-                      }
-                    };
-
-                EndpointWatcher endpointWatcher =
-                    new EndpointWatcherImpl(lrsClient, lrsCallback, localityStore);
-                xdsClient.watchEndpointData(node.getCluster(), endpointWatcher);
-              }
-            }
+    // The endpointWatcher is also updated after switchingLoadBalancer.handleResolvedAddresses().
+    xdsConfig = newXdsConfig;
+  }
 
-            @Override
-            public void shutdown() {
-              if (xdsClient != null) {
-                lrsClient.stopLoadReporting();
-                localityStore.reset();
-                xdsClient.shutdown();
-              }
-            }
-          };
-        }
-      };
+  @Override
+  public void handleNameResolutionError(Status error) {
+    channelLogger.log(ChannelLogLevel.DEBUG, "Name resolution error: '%s'", error);
```

Comment:
should the level be ERROR and not DEBUG since you are reporting an error?

Reply:
Done.

Reply:
I doubt this should be `ERROR` level. As `handleNameResolutionError` is gracefully handling name resolution error encountered by delegating the handling to `switchingLoadBalancer`, although it eventually does nothing (should it really do nothing?). `ERROR` logs are for internal failures that should raise our attention and do something (manual changes) to let it not happen. But here, we are implementing some logic to handle name resolution error (for example, in [`Grpclb`](https://github.com/grpc/grpc-java/blob/09d50932956b550da73bc03a77383d1f3d07c2c6/grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java#L185) if such an error happens, it [turns channel state to transient failure](https://github.com/grpc/grpc-java/blob/09d50932956b550da73bc03a77383d1f3d07c2c6/grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java#L357)). No manual intervention is needed and does not need to raise human attention. I believe `DEBUG` is more appropriate.

Btw, is name resolution error going into black hole really the intended behavior? This method will be triggered by resolver, for example, when LDS/RDS cannot find corresponding resources.

Reply:
`handleNameResolutionError()` is so subtle. I would like to keep the current behavior unchanged in this PR, and may fix that later in a future PR. 

>When LDS/RDS cannot find corresponding resources. 

That's a good point (support you will call `XdsNameResolver.listener.onError()` for LDS/RDS `watcher.onError()`), so I will implement `handleNameResolutionError()` for CDS policy.

I didn't find anything similar to `handleNameResolutionError()` in c-core.

Reply:
Now implemented `handleNameResolutionError()` to go to TRANSIENT_FAILURE if no progress is made previously.